pkg/status/status.go

```diff
@@ -262,6 +269,40 @@ func (s *status) updateStatus(previousStatus *configv1.ClusterOperatorStatus) er
 	return err
 }
 
+// setRelatedObjects populates RelatedObjects in the ClusterOperator.Status.
+// RelatedObjects are consumed by https://github.com/openshift/must-gather
+func (s *status) setRelatedObjects() {
+	// Add the operator's namespace which will result in core resources being gathered
+	objectReferences := []configv1.ObjectReference{
+		{
+			Resource: "namespace",
+			Name:     s.namespace,
+		},
+	}
+
+	// Add the default OperatorSources and it's child resources so that their info is gathered too
+	sources := []string{"redhat-operators", "certified-operators", "community-operators"}
+	for _, source := range sources {
+		objectReferences = append(objectReferences, configv1.ObjectReference{
+			Group:     marketplace.SchemeGroupVersion.Group,
+			Resource:  marketplace.OperatorSourceKind,
+			Namespace: s.namespace,
+			Name:      source,
+		}, configv1.ObjectReference{
+			Group:     marketplace.SchemeGroupVersion.Group,
+			Resource:  marketplace.CatalogSourceConfigKind,
+			Namespace: s.namespace,
+			Name:      source,
+		}, configv1.ObjectReference{
+			Group:     olm.GroupName,
+			Resource:  olm.CatalogSourceKind,
+			Namespace: s.namespace,
+			Name:      source,
+		})
+	}
+	s.clusterOperator.Status.RelatedObjects = objectReferences
+}
+
```

Comment:
What about deployments, services, roles, etc? Are we not including those here?

Reply:
We don't add `Services` and `Roles` for non-private `OperatorSources`. I thought I saw `Deployments` being collected as part of specifying the namespace. Let me check and if it is not I will add that too.

Reply:
`Deployments` are collected by default. Here is an example of the data that is collected:
```
must-gather/
├── cluster-scoped-resources
│   ├── config.openshift.io
│   │   ├── apiservers.yaml
│   │   ├── authentications.yaml
│   │   ├── builds.yaml
│   │   ├── clusteroperators
│   │   │   └── marketplace.yaml
│   │   ├── clusteroperators.yaml
│   │   ├── clusterversions.yaml
│   │   ├── consoles.yaml
│   │   ├── dnses.yaml
│   │   ├── featuregates.yaml
│   │   ├── images.yaml
│   │   ├── infrastructures.yaml
│   │   ├── ingresses.yaml
│   │   ├── networks.yaml
│   │   ├── oauths.yaml
│   │   ├── projects.yaml
│   │   ├── schedulers.yaml
│   │   └── support.yaml
│   └── operator.openshift.io
└── namespaces
    └── openshift-marketplace
        ├── apps
        │   ├── daemonsets.yaml
        │   ├── deployments.yaml
        │   ├── replicasets.yaml
        │   └── statefulsets.yaml
        ├── apps.openshift.io
        │   └── deploymentconfigs.yaml
        ├── autoscaling
        │   └── horizontalpodautoscalers.yaml
        ├── batch
        │   ├── cronjobs.yaml
        │   └── jobs.yaml
        ├── build.openshift.io
        │   ├── buildconfigs.yaml
        │   └── builds.yaml
        ├── core
        │   ├── configmaps.yaml
        │   ├── events.yaml
        │   ├── pods.yaml
        │   ├── replicationcontrollers.yaml
        │   ├── secrets.yaml
        │   └── services.yaml
        ├── image.openshift.io
        │   └── imagestreams.yaml
        ├── openshift-marketplace.yaml
        ├── operators.coreos.com
        │   ├── catalogsourceconfigs
        │   │   ├── certified-operators.yaml
        │   │   ├── community-operators.yaml
        │   │   └── redhat-operators.yaml
        │   ├── catalogsources
        │   │   ├── certified-operators.yaml
        │   │   ├── community-operators.yaml
        │   │   └── redhat-operators.yaml
        │   └── operatorsources
        │       ├── certified-operators.yaml
        │       ├── community-operators.yaml
        │       └── redhat-operators.yaml
        ├── pods
        │   ├── certified-operators-7755565bd7-5hnzn
        │   │   ├── certified-operators
        │   │   │   └── certified-operators
        │   │   │       ├── healthz
        │   │   │       └── logs
        │   │   │           ├── current.log
        │   │   │           └── previous.log
        │   │   └── certified-operators-7755565bd7-5hnzn.yaml
        │   ├── community-operators-c944ffc4f-w9r6l
        │   │   ├── community-operators
        │   │   │   └── community-operators
        │   │   │       ├── healthz
        │   │   │       └── logs
        │   │   │           ├── current.log
        │   │   │           └── previous.log
        │   │   └── community-operators-c944ffc4f-w9r6l.yaml
        │   ├── marketplace-operator-67775478d7-vcjxv
        │   │   ├── marketplace-operator
        │   │   │   └── marketplace-operator
        │   │   │       ├── healthz
        │   │   │       └── logs
        │   │   │           ├── current.log
        │   │   │           └── previous.log
        │   │   └── marketplace-operator-67775478d7-vcjxv.yaml
        │   └── redhat-operators-6f4b66f8d-vdwv9
        │       ├── redhat-operators
        │       │   └── redhat-operators
        │       │       ├── healthz
        │       │       └── logs
        │       │           ├── current.log
        │       │           └── previous.log
        │       └── redhat-operators-6f4b66f8d-vdwv9.yaml
        └── route.openshift.io
            └── routes.yaml
```

`healthz` data is not collected as `must-gather` assumes HTTP liveness while `operator-registry` uses a command.